personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+import sys, os, warnings, getopt
+import datetime, string, random
+import openpyxl
+from tabulate import tabulate
+
+
+def extract_data_from_source(dir_path):
```

Comment:
This function can be broken into 3 parts:
1. First function opens the xls file and reads the data and just appends the rows to a list (this is helpful so that during testing, we can skip xls file creation and test the next two functions or the entire code using a list of txns)
2. Iterates over the transactions in this list, and then creates the transaction dictionary
3. Iterates over the transactions in this list, and then creates the monthly transaction dictionary